### ISSUE TITLE
Warnings when generating software quality reports in apps

### ISSUE THREAD

**Issue by bwspenc** [idaholab/moose#17673]
## Bug Description
In the software quality reports generated by apps, we're getting warnings like this when we point to the main `sqa/inl_records.md` file, even though the links to that are correctly generated:

```
Failed to locate 'safety_software_determination' document: sqa/inl_records.md#safety_software_determination
Control this message using: 'log_safety_software_determination'
```

## Steps to Reproduce
The pages currently generated by apps all have this issue, and as a result, we're still enabling warnings for these.  See this page for BlackBear for an example:
https://mooseframework.inl.gov/blackbear/sqa/index.html

## Impact
This is mostly a nuisance, although we do want to show a clean build with no warnings or errors for the upcoming audit.

**Comment by bwspenc** [comment]
FYI @novasr

**Comment by aeslaughter** [comment]
The fix requires that utilities in python/moosesqa be able to load in more that one location for searching files. I didn't design it to look for documents outside itself.

**Comment by aeslaughter** [comment]
This was fixed in #17715
